packages/adp-tooling/src/base/validate.ts

```diff
@@ -45,19 +47,25 @@ export default class UI5Validator {
      * Checks if middleware has all the necessary properties an adaptation project needs.
      *
      * @param customConfiguration AdpCustomConfiguration
-     * @param previewMiddleware CustomMiddleware
+     * @param fioriPreview CustomMiddleware
+     * @param fioriProxy CustomMiddleware
      */
     private static checkMiddlewareProperties(
         customConfiguration: AdpCustomConfiguration | undefined,
-        previewMiddleware: CustomMiddleware<PreviewMiddlewareConfiguration> | undefined
+        fioriPreview: CustomMiddleware<FioriToolsPreviewConfig> | undefined,
+        fioriProxy: CustomMiddleware<FioriToolsProxyConfig> | undefined
     ): void {
-        if (!customConfiguration || !previewMiddleware) {
+        if (!customConfiguration || !fioriPreview || !fioriProxy) {
             throw new Error('Missing required custom middleware or custom configuration in ui5.yaml');
         }
         this.assertProperties(['environment'], customConfiguration);
-        this.assertProperties(['configuration'], previewMiddleware);
-        this.assertProperties(['adp'], previewMiddleware.configuration);
-        this.assertProperties(['target'], previewMiddleware.configuration.adp);
-        this.assertProperties(['url', 'client'], previewMiddleware.configuration.adp.target);
+        this.assertProperties(['configuration'], fioriPreview);
+        this.assertProperties(['adp'], fioriPreview.configuration);
+        this.assertProperties(['target'], fioriPreview.configuration.adp);
+        this.assertProperties(['url', 'client'], fioriPreview.configuration.adp.target);
```

Comment:
When the user creates an ADP in BAS with the new structure, he will not have a url and client properties, which will throw an error and fail. The same will happen with the destination property on VS Code.

Reply:
There is still no BAS implementation, and we are not populating destination in ui5.yaml if a project is generated from BAS